frontend/libs/studio-components/src/components/StudioSpinner/StudioSpinner.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { HTMLAttributes, forwardRef, useId } from 'react';
+import { Paragraph, Spinner } from '@digdir/design-system-react';
+import type { SpinnerProps } from '@digdir/design-system-react';
+import { useTranslation } from 'react-i18next';
+
+export type StudioSpinnerProps = {
+  spinnerText?: string;
+} & Pick<SpinnerProps, 'size' | 'variant'> &
+  HTMLAttributes<HTMLDivElement>;
+
+export const StudioSpinner = forwardRef<HTMLDivElement, StudioSpinnerProps>(
+  ({ spinnerText, size = 'medium', variant = 'interaction', ...rest }, ref): JSX.Element => {
+    const { t } = useTranslation();
+
+    const spinnerDescriptionId = useId();
+
+    // TODO - Wrap with CENTER
+    return (
+      <div ref={ref} {...rest}>
+        <Spinner
+          title={!spinnerText && t('general.loading')}
+          size={size}
+          variant={variant}
+          aria-describedby={spinnerText && spinnerDescriptionId}
+          data-testid='studio-spinner-test-id'
+        />
+        {spinnerText && (
+          <Paragraph as='div' id={spinnerDescriptionId}>
+            {spinnerText}
+          </Paragraph>
+        )}
+      </div>
+    );
+  },
+);
+
+StudioSpinner.displayName = 'StudioSpinner';
```

Comment:
One more, do we need the displayName set here? 🤔

Reply:
Nope 😄